### ISSUE TITLE
Identifications of cells

### ISSUE THREAD

**Issue by gerhardol** [lordgreg/Sfen#30]
For me, psc is always -1. I have mostly 3G around I believe, but psc does not seem to be available.
Samsung Galaxy S2 with CM11
psc is for sure always -1 in 2g/GSM
I believe that psc should not be included for identifying with GsmCellLocation. For sure not when -1, as that should trigger the isError()
psc is scrambling code, not sure if that is applicable to identify sectors for instance. 

On the other hand, I believe Lac should be included for CellIdentityGsm/Wcdma too, in addition to Mnc/Mcc/Cid.
LTE probably ok with Mnc/Mcc/Ci.

Open as an issue rather than a patch, as you likely have done more research than me, but maybe not see the psc=-1 issue

**Comment by mvglasow** [comment]
> It seems like Psc is set in some phones. Maybe it is the HW. But at least my SGS2 w CM11 always have -1.

Yes, Samsung phones (at least of that generation) are known not to report the PSC, or any neighboring cells. This is a limitation of the firmware.

A list of the limitations of a few devices can be found at https://docs.cellmapper.net/mw/index.php/Tested_Phones

> I believe that psc should not be included for identifying with GsmCellLocation.

+1, the MCC/MNC/LAC/CID tuple are unique (at least within the same network type—not sure if collisions between, say, a 2G cell and a 3G cell are possible). PSC is a local identifier—no two neighboring cells can have the same PSC, and no cell can have two neighbors with the same PSC. Some have suggested that carriers occasionally change the PSCs for a cell (while keeping the CID), but I have no reliable information on that.

> I do not know why I get CellLocation rather than CellInfo too.

Again, limitation of the phone firmware. There are three different API calls, and not all phones implement all of them.

> There should be a way to get the antenna sectors too, is it hidden in the lac/cid.

Some carriers indeed encode antenna sectors in the CID (the LAC is an area code, thus different antenna sectors of the same tower would normally share the same LAC). However, this is purely by convention and entirely at the whim of the carrier—schemes vary from carrier to carrier, and are subject to change without notice.
